api/src/main/java/org/pmiops/workbench/cohortbuilder/querybuilder/FactoryKey.java

```diff
@@ -0,0 +1,39 @@
+package org.pmiops.workbench.cohortbuilder.querybuilder;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * This enum maps specific implementations of {@link AbstractQueryBuilder} to specific
+ * criteria types/searches. This enum is built specifically for use of the
+ * {@link org.pmiops.workbench.cohortbuilder.QueryBuilderFactory} to get the correct
+ * factory implementation.
+ */
+public enum FactoryKey {
+    CRITERIA("criteria-type",  Arrays.asList("ICD9-TREE", "ICD10-TREE", "CPT-TREE")),
+    CODES("codes-type", Arrays.asList("ICD9", "ICD10", "CPT")),
+    /** TODO: this is temporary and will be removed when we figure out the conceptId mappings **/
+    GROUP_CODES("group-codes-type", Arrays.asList("GROUP_CODES")),
+    DEMO("demo-type", Arrays.asList("DEMO"));
+
+    private String name;
+    private List<String> types;
+
+    private FactoryKey(String name, List<String> types) {
+        this.name = name;
+        this.types = types;
+    }
+
+    public String getName() {
+        return this.name;
+    }
+
+    public static String getKey(String type) {
+        for (FactoryKey key: values()) {
+            if (key.types.contains(type)) {
+                return key.getName();
+            }
+        }
+        return CRITERIA.getName();
```

Comment:
Should this throw IllegalArgumentException instead? (If the type doesn't match anything that seems like a bug...)

Reply:
Yes... agreed that this is a bug.